spec/requests/posts_spec.rb

```diff
@@ -1,11 +1,33 @@
 require 'rails_helper'
 
-RSpec.describe "Posts", type: :request do
-  describe "GET /index" do
-    it "returns http success" do
-      get "/posts/index"
-      expect(response).to have_http_status(:success)
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:example) { get('/users/1/posts') }
+    it 'should return success status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'it renders index template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'should have a body' do
+      expect(response.body).to include('<h1> Posts index</h1>')
     end
   end
 
-end
+  describe 'GET /show' do
+    before(:example) { get('/users/1/posts/1') }
+    it 'should return success status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'it renders index template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'should have a body' do
+      expect(response.body).to include('<h1>Posts show</h1>')
+    end
+  end
+end
```

Comment:
- [ ] There are 3 tests that didn't pass successfully, Please fix that.
It may be working on your side because you have data on your local database.

![image](https://github.com/mudasiri/mybloggerapp/assets/68796232/701c6ac7-ccac-46c6-a824-6c74ecddfe42)